core/src/main/scala/commandcenter/command/OpacityCommand.scala

```diff
@@ -18,8 +21,12 @@ final case class OpacityCommand() extends Command[Unit] {
 
   val opacityCommand = decline.Command("opacity", title)(opacity)
 
-  def preview(searchInput: SearchInput): ZIO[Env, CommandError, List[PreviewResult[Unit]]] =
+  def preview(searchInput: SearchInput): ZIO[Env, CommandError, List[PreviewResult[Unit]]] = {
+    val isSwingTerminal = searchInput.context.terminal.terminalType == TerminalType.Swing
     for {
+      _              <- effectBlocking(isSwingTerminal && !GraphicsUtil.isOpacitySupported)
+                          .mapError(CommandError.UnexpectedException)
+                          .whenM(ZIO.fail(CommandError.NotApplicable))
```

Comment:
This part looks backwards to me. The condition part is set to always fail, which means this operation can never succeed.

It should look something like:

```scala
ZIO.fail(CommandError.NotApplicable).whenM(...)
```

or some people prefer the syntax the other way around:

```scala
ZIO.whenM(...)(ZIO.fail(CommandError.NotApplicable))
```

Reply:
Fixed. I've decided to use `unlessM` to make it more natural.